### PR TITLE
Fixes #2603 by defining log templates and binding variables.

### LINKED ISSUE #2603: ActorLogging warning does not expand Throwable
`log.warning(message, throwable)` is not handled the way other logging frameworks handle that and lead to wrong log messages like: `WARNING arguments left: 1`

### PR DESCRIPTION
**Pull request by aquamatthias**
After merge cherry pink on master.

### CODE REVIEW

File: src/main/scala/mesosphere/marathon/event/http/HttpEventActor.scala [hunk 1]
```diff
@@ -45,10 +44,9 @@ class HttpEventActor(val subscribersKeeper: ActorRef) extends Actor with ActorLo
     response.onComplete {
       case Success(res) =>
         if (res.status.isFailure)
-          log.warning(s"Failed to post $event to $urlString")
-
+          log.warning(s"Post {} to {} resulted in error response {}", event, urlString, res.status)
```
Comment (meichstedt) [inline src/main/scala/mesosphere/marathon/event/http/HttpEventActor.scala]: I suppose your point was to replace scala string interpolation with the log.warning varargs approach – you missed removing the string interpolation prefix that will lead to a call to `StringContext.s()``

File: src/main/scala/mesosphere/marathon/event/http/HttpEventStreamHandleActor.scala [hunk 2]
```diff
@@ -76,11 +76,11 @@ class HttpEventStreamHandleActor(
       //Do not act any longer on any event.
       context.become(Actor.emptyBehavior)
     case _ =>
-      log.warning(s"Could not send message to $handle", ex)
+      log.warning("Could not send message to {} reason: {}", handle, ex)
   }
 
   private[this] def dropEvent(event: MarathonEvent): Unit = {
-    log.warning(s"Ignore event $event for handle $handle (slow consumer)")
+    log.warning("Ignore event {} for handle {} (slow consumer)", event, handle)
```
Comment (meichstedt) [inline src/main/scala/mesosphere/marathon/event/http/HttpEventStreamHandleActor.scala]: Is there a specific reason you chose varargs substitution over the scala way? As far as I can see, doing it this way and adding another arg to the list will still result in a `WARNING arguments left: 1` error.

**Comment by meichstedt** [comment]
See my comments above. I am not sure, whether we should use varargs here or go for scala string interpolation in general – I think we use of string interpolation mor often than not, so we might do here as well?

File: src/main/scala/mesosphere/marathon/event/http/HttpEventStreamHandleActor.scala (hunk 2 above)
Comment (aquamatthias) [inline src/main/scala/mesosphere/marathon/event/http/HttpEventStreamHandleActor.scala]: Looked like the default usage in our code base. 
My preference would be to use a clever scala logging system, where we don't have to decide which way to use...

**Pull request merged**